src/api/app/models/project.rb

```diff
@@ -350,7 +350,7 @@ def self.get_maintenance_project(at = nil)
     at ||= AttribType.find_by_namespace_and_name!('OBS', 'MaintenanceProject')
     maintenanceProject = Project.find_by_attribute_type(at).first
     unless maintenanceProject and check_access?(maintenanceProject)
-      raise UnknownProject.new 'There is no project flagged as maintenance project on server and no target in request defined.'
+      raise BsRequestAction::UnknownProject.new 'There is no project flagged as maintenance project on server and no target in request defined.'
```

Comment:
I would try to use UnknownObjectError in first place, since we use that one also in other places of the project model and it exists there. It is a method which finds a certain object like get_by_name and we use this exception there as well.

My point was not to avoid exceptions, but to avoid a "virtual relationship" to a request which might not even exist. (we use the method also in non-request related code)